src/main/riddl/Events/eventContext.riddl

```diff
@@ -2,195 +2,666 @@ type EventId is Id (ImprovingApp.Events.EventContext.Event)
   briefly "Unique identifier for Tenant Entity"
 
 context EventContext is {
-    include "eventMessages"
-    include "eventTypeDefinitions"
-    include "eventSources"
+  include "eventMessages"
+  include "eventTypeDefinitions"
+  include "eventSources"
 
-    entity Event is {
-        option is aggregate
+  function incrementDelayedEventTime is {
+    requires {originalTime: DateTime, duration: Duration}
+    returns { newTime: DateTime }
+  } briefly "used for incrementing start time and end time when DelayEvent is received"
 
-        handler EventHandler is {
-            on command ScheduleEvent {
-                then morph entity Event to state Scheduled with
-                  !Event.ScheduledState(
-                    eventInfo = !EventInfo(),
-                    eventMetaInfo = !EventMetaInfo(),
-                    reservation = !ReservationId()
-                  )
-            }
-            on other {
-            then error "You must first create an event using ScheduleEvent command."
-            }
-        }
+  function startTimeIfPast is {
+    requires {startTime: DateTime}
+    returns { timeOpt: DateTime? }
+  } briefly "returns startTime if it is past, empty option if it isn't. This is used when CancelEvent is received"
+
+  function nowIfStartIsPast is {
+    requires {startTime: DateTime}
+    returns { nowOpt: DateTime? }
+  } briefly "returns now if startTime is past, empty option if it isn't. This is used when CancelEvent is received"
+
+  function newEditableInfoFromFieldsPresentInEditable is {
+    requires { editable: EditableEventInfo, oldInfo: EditableEventInfo }
+    returns { newInfo: EditableEventInfo }
+  } briefly "returns newInfo with all fields present in editable and all those not present there taken from oldInfo"
+
+  function newInfoFromFieldsPresentInEditable is {
+    requires { editable: EditableEventInfo, oldInfo: EventInfo }
+    returns { newInfo: EventInfo }
+  } briefly "returns newInfo with all fields present in editable and all those not present there taken from oldInfo"
+
```

Comment:
I am not sure of the purpose of the above functions. I am sure it will make more sense when I get to the context of where they are used. But just reading through the changes I am not sure why we need these.